Lib/ruby/rubywstrings.swg

```diff
@@ -8,23 +8,23 @@
 SWIGINTERN int
 SWIG_AsWCharPtrAndSize(VALUE obj, wchar_t **cptr, size_t *psize, int *alloc)
 {
+  static rb_encoding* wstr_enc = rb_to_encoding(rb_str_new_cstr( SWIG_RUBY_WSTRING_ENCODING ));
```

Comment:
That may work in C++ (though could have issues if multiple threads can execute this concurrently), but in C a `static` variable needs a constant initialiser.  The usual idiom if to make the static initialiser `NULL` and then to check at runtime, something like: `if (wstr_enc == NULL) wstr_enc = rb_to_encoding(rb_str_new_cstr( SWIG_RUBY_WSTRING_ENCODING ));`

But again that's potentially racy when multi-threading.

So perhaps it's best to find a way to get the encoding more efficiently.

For UTF-8, it looks like `rb_utf8_encoding()` might be a better option (it returns `rb_encoding *`).

For UTF-16/UTF-32 perhaps `rb_enc_find_index()` (takes `const char *`) then `rb_enc_from_index()` to get the `rb_encoding *`?

But I'm not really familiar with these APIs, so maybe these are still too slow.